dag/set_test.go

```diff
@@ -119,3 +119,13 @@ func TestSetCopy(t *testing.T) {
 	}
 
 }
+
+func BenchmarkSetIntersectionLargeToSmall(b *testing.B) {
+	var small, large Set
+	for i := 0; i < 1000000; i++ {
+		large.Add(i)
+	}
+	small.Add(1)
+	b.ResetTimer()
+	large.Intersection(small)
```

Comment:
A benchmark measures the code to be executed as is it ran `b.N` number of times. I think just wrapping the `Intersection` line in the for loop should suffice here.